pandas/core/generic.py

```diff
@@ -5601,53 +5601,83 @@ def clip(self, lower=None, upper=None, axis=None, inplace=False,
         """
         Trim values at input threshold(s).
 
+        Elements above/below the upper/lower thresholds will be changed to
+        upper/lower thresholds.
+
         Parameters
         ----------
-        lower : float or array_like, default None
-        upper : float or array_like, default None
-        axis : int or string axis name, optional
-            Align object with lower and upper along the given axis.
+        lower : float, array-like or None, default None
+            Lower threshold for clipping.  Values smaller than `lower` will be
+            converted to `lower`.
+        upper : float, array-like or None, default None
+            Upper threshold for clipping.  Values larger than `upper` will be
+            converted to `upper`.
+        axis : {0 or 'index', 1 or 'columns', None}, default None
+            Apply clip by index (i.e. by rows) or columns.
         inplace : boolean, default False
             Whether to perform the operation in place on the data
-                .. versionadded:: 0.21.0
+                .. versionadded:: 0.21.0.
+        *args : Additional keywords have no effect but might be accepted
+            for compatibility with numpy.
+        **kwargs : Additional keywords have no effect but might be accepted
+            for compatibility with numpy.
 
         Returns
         -------
-        clipped : Series
+        `Series` or `DataFrame`.
+            DataFrame is returned with those values above/below the
+            `upper`/`'lower` thresholds set to the threshold values.
+
+        See Also
+        --------
+        pandas.Series.clip : Trim values at input threshold(s).
 
```

Comment:
I think the prefix `pandas` is not needed, just `Series.clip`. Also, not sure if that was already discussed, but don't we want `clip_lower` and `clip_upper` here?

Reply:
I think you previously mentioned that those were generic.  Did we want them included still?

Reply:
Sorry, I think I wasn't clear enough. You should check it, but as this docstring is in `generic.py` I assume it'll be used by both `Series` and `DataFrame`. That's why I said it was (the docstring) was generic.

So, as your assigned docstring was `DataFrame.clip` (and it's what it's in the title of the PR) but you're actually working also in `Series.clip`, I don't think it makes sense to only add `Series.clip` as @jreback suggested. I'd say that we probably want also `DataFrame.clip`. Even if it'll be a bit weird to have in the `See Also` the same method which is being documented, but I think it's all right. In a separate PR we could use a template and just have the right one.

So, summarizing, the see also should contain `Series.clip`, `DataFrame.clip`, and the `clip_lower` and `clip_upper` for each of them (I assume both classes have it). And assuming that `clip` is commonly used with `quantile` as in your example (I don't know), I would also add it.

@jreback do you agree?

Reply:
Ah, yes I see, I was mistaken.  Sure, I can add all those.